File: 1350-remove-sub-folders-from-the-filesystem/solution.cpp
class Solution {
public:
    vector<string> removeSubfolders(vector<string>& folder) {
        sort(folder.begin(), folder.end()); // lexiography 

        vector<string> ans;
        ans.push_back(folder[0]);

        for (int i = 1; i < folder.size(); ++i) {
            string s = ans.back();
            s.push_back('/');

            // compare current folder with last added folder
            // cjecl of folder[i] starts with lastFolder
            if (folder[i].compare(0, s.size(), s) != 0) {
                // its not a subfolder
                ans.push_back(folder[i]);
            }
        }

        return ans;
    }
};

File: 1350-remove-sub-folders-from-the-filesystem/solution.py
class Solution:
    def removeSubfolders(self, folder: List[str]) -> List[str]:
        folder.sort() # lexiographically

        res = [folder[0]] # add the first folder

        # iterate through each folder and check if sub folder exists
        for i in range(1, len(folder)): # 1 -> n-1
            last = res[-1]
            last += "/"

            if not folder[i].startswith(last):
                res.append(folder[i])

        return res  
